File: chatwoot/java/src/main/java/oseg/chatwoot_examples/UpdateAgentsInInboxExample.java
package oseg.chatwoot_examples;

import com.chatwoot.client.ApiException;
import com.chatwoot.client.Configuration;
import com.chatwoot.client.api.*;
import com.chatwoot.client.auth.*;
import com.chatwoot.client.JSON;
import com.chatwoot.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class UpdateAgentsInInboxExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("userApiKey")).setApiKey("USER_API_KEY");

        var addNewAgentToInboxRequest = new AddNewAgentToInboxRequest();
        addNewAgentToInboxRequest.inboxId("inbox_id_string");
        addNewAgentToInboxRequest.userIds(List.of ());

        try
        {
            var response = new InboxesApi(config).updateAgentsInInbox(
                0, // accountId
                addNewAgentToInboxRequest // data
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling InboxesApi#updateAgentsInInbox");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: chatwoot/kotlin/src/main/kotlin/oseg/chatwoot_examples/UpdateAgentsInInboxExample.kt
package oseg.chatwoot_examples

import com.chatwoot.client.infrastructure.*
import com.chatwoot.client.apis.*
import com.chatwoot.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class UpdateAgentsInInboxExample
{
    fun updateAgentsInInbox()
    {
        ApiClient.apiKey["api_access_token"] = "USER_API_KEY"

        val addNewAgentToInboxRequest = AddNewAgentToInboxRequest(
            inboxId = "inbox_id_string",
            userIds = listOf (),
        )

        try
        {
            val response = InboxesApi().updateAgentsInInbox(
                accountId = 0,
                _data = addNewAgentToInboxRequest,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling InboxesApi#updateAgentsInInbox")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling InboxesApi#updateAgentsInInbox")
            e.printStackTrace()
        }
    }
}
